build/Dockerfile

```diff
@@ -20,8 +21,7 @@ RUN apk add --no-cache libcap \
 ############################################# Base image for Debian with NGINX Plus #############################################
 FROM debian:buster-slim AS debian-plus
 ARG IC_VERSION
-ENV NGINX_PLUS_VERSION 23-1~buster
-ENV NGINX_NJS_VERSION 23+0.5.0-1~buster
+ARG NGINX_PLUS_VERSION
```

Comment:
I wonder if should provide a fallback option - allow customers to use specific version. 
For example, imagine a problem is discovered in a new version of NGINX Plus patch version. Or in a module. Would it be possible for a customer to fall back to the previous versions?

Reply:
hmm has this ever happened? I think if there was a known problem with a specific version of NGINX Plus, the team would release a new version right away?

I can't think of a way to allow for a specific version with this method, you can only change R version I think, like use `r23`. In Debian the patch version is not specified in the virtual package
```
Provides:
24-2~buster - nginx-plus-r24 ...
24-1~buster - nginx-plus-r24 ...
```
they're both `nginx-plus-r24`. 

I remember reading that this is because `apt` was made with the assumption that people want to always install the latest version of a package 99% of the time .

Reply:
hm. perhaps it's just a matter of some additional documentation. For example:

```bash
	&& apt-get install --no-install-recommends --no-install-suggests -y \
	## the commands below install the latest patch version of the major NGINX Plus version specified in $NGINX_PLUS_VERSION
	## as well as the latest versions of the modules packages compatible with $NGINX_PLUS_VERSION
	## if you would like to fix the installed version, modify the lines below
	## For example:
	##  nginx-plus=24-2~buster
	nginx-plus=$(apt-cache showpkg nginx-plus-$NGINX_PLUS_VERSION | awk '{ if ($1 == "nginx-plus") {print $2; exit}}') \
	nginx-plus-module-njs=$(apt-cache showpkg nginx-plus-module-njs-$NGINX_PLUS_VERSION | awk '{ if ($1 == "nginx-plus-module-njs") {print $2; exit}}') \
```

I think something like that will give a clear way for customers to fix the versions if that is required. Whether they want reproducible builds or need to roll back to the previous NGINX Plus or any module version

What do you think?

Reply:
I've changed it to use `nginx-plus-r24` to match the other builds.

Reply:
what would be a way to fix the versions? could we mention how to do that?

Reply:
Talked to @brianehlert and we shouldn't block this PR because of docs. If a request for docs on how to change the version arises, we'll update them then.

Reply:
sounds good. could you address this one https://github.com/nginxinc/kubernetes-ingress/pull/1604#discussion_r634786314 ?
we have the same thing for the app protect images as well - because now we will always install the latest

Reply:
The major version is set in the dockerfile.
The convenience is that the latest matching release of the major version is installed.
I believe this aligns with expectations and provides reasonable stability.

I would expect app protect to operate the same way.  Major version pin.  Latest patch.
This should avoid problems.
I would expect app protect to pin to versions of nginx.  As there are modules involved and those are compiled to the nginx binary.

Adding a note to rebuild your image when there is an update to nginx to get the latest update is prudent.